### PR TITLE
Implemented redux without saga

### PR DESCRIPTION
**Pull request by dfcz3322**

### CODE REVIEW

File: src/actions/comics.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { IMarvelEntityResponse } from "../interface/interface";
+
+export const setComics = (comics: IMarvelEntityResponse[]): {type: string, payload: IMarvelEntityResponse[]} => {
+    return {
+        type: 'comics/setComics',
```
Comment (alinalaktionova) [inline src/actions/comics.ts]: types should be constants and uppercased

File: src/actions/comics.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { IMarvelEntityResponse } from "../interface/interface";
```
Comment (alinalaktionova) [inline src/actions/comics.ts]: no need to create separate file for one action

File: src/reducer/comicsReducer.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+export interface IComicsReduxState {
+    comics: IMarvelEntityResponse[];
+}
+
+const initialComics: IComicsReduxState = {
+    comics: []
+};
+
+export const comics = (state: IComicsReduxState = initialComics, action: { type: string, payload: unknown }): IComicsReduxState => {
```
Comment (alinalaktionova) [inline src/reducer/comicsReducer.tsx]: don't use type unknown

File: src/reducer/searchReducer.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    characters: [],
+    searchQuery: ""
+};
+
+
+export const search = (state: ISearchReduxState = initialSearch, action: { type: string, payload: unknown }): ISearchReduxState => {
+    switch(action.type) {
+        case 'search/setCharacters':
+            return {...state, characters: action.payload as IMarvelEntityResponse[] }
+        case 'search/setSearchQuery':
```
Comment (alinalaktionova) [inline src/reducer/searchReducer.tsx]: what is the point of storing search query in redux?

File: src/pages/search.tsx [hunk 5]
```diff
@@ -6,34 +6,45 @@ import { SearchLogo } from "../components/searchLogo/searchLogo";
...
-        const searchParam = new URLSearchParams(this.props.location.search).get("search");
-        this.state = { characters: [], searchQuery: searchParam ?? "", isLoading: false };
+        this.state = { isLoading: false };
     }
 
-    getHero(): void {
+    getHero(searchParam = ""): void {
         this.setState({ isLoading: true });
-        handleRequest(`characters`, this.state.searchQuery)
+        handleRequest(`characters`, searchParam)
```
Comment (alinalaktionova) [inline src/pages/search.tsx]: '' instead ``

**Review by alinalaktionova — CHANGES_REQUESTED** [review]
your code style doesn't follow common js conventions. Edit spacing through the code and replace double quotes on single

File: src/reducer/searchReducer.tsx (hunk 4 above)
Comment (dfcz3322) [reply src/reducer/searchReducer.tsx]: this is in order to keep consistency as now characters are in redux store so I wanted to move as much as possible to redux store and not to mix component state and redux state. I can revert if it's incorrect

Comment (alinalaktionova) [reply src/reducer/searchReducer.tsx]: you can just past it as function argument and not store

**Review by alinalaktionova — COMMENTED** [review]
remove search query from redux

File: src/actions/search.ts [hunk 6]
```diff
@@ -3,11 +3,6 @@ interface ISetCharacters {
   type: string;
   payload: IMarvelEntityResponse[];
 }
-interface ISetSearchQuery {
-  type: string;
-  payload: string;
-}
-
 export const SET_CHARACTERS = 'SEARCH/SET_CHARACTERS';
```
Comment (alinalaktionova) [inline src/actions/search.ts]: just SET_CHARACTERS

File: src/actions/search.ts [hunk 7]
```diff
@@ -3,11 +3,6 @@ interface ISetCharacters {
...
 export const SET_SEARCH_QUERY = 'SEARCH/SET_SEARCH-QUERY';
```
Comment (alinalaktionova) [inline src/actions/search.ts]: I think it's redundant for now

File: src/pages/search.tsx [hunk 8]
```diff
@@ -42,20 +41,20 @@ class Search extends React.Component<ISearchProps, ISearchState> {
...
-      this.props.setSearchQuery(searchParam);
+      this.setState({searchQuery: searchParam});
     }
     this.getHero(searchParam as string);
   }
   onSearch = (query: string): void => {
     this.props.history.push({
       pathname: '/',
-      search: `?search=${query}`,
+      ...(query.length && {search: `?search=${query}`}),
```
Comment (alinalaktionova) [inline src/pages/search.tsx]: not readable

**Review by alinalaktionova — CHANGES_REQUESTED** [review]

**Review by alinalaktionova — APPROVED** [review]

**Pull request merged**